src/test/java/testsuite/integration/container/aurora/AuroraMysqlReadWriteSplittingTest.java

```diff
@@ -523,6 +523,85 @@ public void test_multiHostUrl_topologyOverridesHostList() throws SQLException {
     }
   }
 
+  @Test
+  public void test_readerLoadBalancing_autocommitTrue() throws SQLException {
+    final String initialWriterId = instanceIDs[0];
+
+    Properties props = getPropsWithReadWritePlugin();
+    props.setProperty(PropertyKey.loadBalanceReadOnlyTraffic.getKeyName(), "true");
+    try (final Connection conn = connectToInstance(initialWriterId + DB_CONN_STR_SUFFIX, MYSQL_PORT, props)) {
+      String writerConnectionId = queryInstanceId(conn);
+      assertEquals(initialWriterId, writerConnectionId);
+      assertTrue(isDBInstanceWriter(writerConnectionId));
+
+      conn.setReadOnly(true);
+      String previousReaderId = queryInstanceId(conn);
+      assertNotEquals(writerConnectionId, previousReaderId);
+      assertTrue(isDBInstanceReader(previousReaderId));
+
+      for (int i = 0; i < 5; i++) {
+        String nextReaderId = queryInstanceId(conn);
+        assertNotEquals(previousReaderId, nextReaderId);
+        assertTrue(isDBInstanceReader(previousReaderId));
+        previousReaderId = nextReaderId;
+      }
+
+      for (int i = 0; i < 5; i++) {
+        Statement stmt = conn.createStatement();
+        stmt.executeQuery("SELECT " + i);
+
+        ResultSet rs = stmt.getResultSet();
+        rs.next();
+        assertEquals(i, rs.getInt(1));
+      }
+    }
+  }
+
+  @Test
+  public void test_readerLoadBalancing_autocommitFalse() throws SQLException {
+    final String initialWriterId = instanceIDs[0];
+
+    Properties props = getPropsWithReadWritePlugin();
+    props.setProperty(PropertyKey.loadBalanceReadOnlyTraffic.getKeyName(), "true");
+    try (final Connection conn = connectToInstance(initialWriterId + DB_CONN_STR_SUFFIX, MYSQL_PORT, props)) {
+      String writerConnectionId = queryInstanceId(conn);
+      assertEquals(initialWriterId, writerConnectionId);
+      assertTrue(isDBInstanceWriter(writerConnectionId));
+
+      conn.setReadOnly(true);
+      conn.setAutoCommit(false);
+      String previousReaderId = queryInstanceId(conn);
+      assertNotEquals(writerConnectionId, previousReaderId);
+      assertTrue(isDBInstanceReader(previousReaderId));
+
+      for (int i = 0; i < 5; i++) {
+        conn.commit();
```

Comment:
Do we need a statement that opens transaction?

Reply:
Sure, I'll add that in just in case